book/src/granular_lint_groups.md

```diff
@@ -0,0 +1,219 @@
+
+```
+#![warn(
+    clippy::format_push_string,
+    clippy::missing_asserts_for_indexing
+)]
+```
+</details>
+
+<details>
+<summary>Lint table</summary>
+
+```
+[lints.clippy]
+format_push_string = "warn"
+missing_asserts_for_indexing = "warn"
+```
+</details>
+<!-- lint-group-end: perf-restriction -->
+
+## Perf-nursery
+
+These are `nursery` lints that either were previously in the `perf` category or are intended to be in `perf`
+but have too many false positives.
+Some of them may also be simply wrong in certain situations and end up slower,
+so you should make sure to read the description to learn about possible edge cases.
+
+<!-- lint-group-start: perf-nursery -->
+Lints: `redundant_clone`, `iter_with_drain`, `mutex_integer`, `or_fun_call`, `significant_drop_tightening`, `trivial_regex`, `needless_collect`
+
+<details>
+<summary>#![warn] attribute</summary>
+
+```
+#![warn(
+    clippy::redundant_clone,
+    clippy::iter_with_drain,
+    clippy::mutex_integer,
+    clippy::or_fun_call,
+    clippy::significant_drop_tightening,
+    clippy::trivial_regex,
+    clippy::needless_collect
+)]
+```
+</details>
+
+<details>
+<summary>Lint table</summary>
+
+```
+[lints.clippy]
+redundant_clone = "warn"
+iter_with_drain = "warn"
+mutex_integer = "warn"
+or_fun_call = "warn"
+significant_drop_tightening = "warn"
+trivial_regex = "warn"
+needless_collect = "warn"
+```
+</details>
+<!-- lint-group-end: perf-nursery -->
+
+## Panicking
+
+These are `restriction` lints that look for patterns that can introduce panics.
+
+Usually panics are not something that one should want to avoid and most of the time panicking is perfectly valid
+(hence why these lints are allow-by-default),
+but users may want to forbid any use of panicky functions altogether in specific contexts.
+
+One use case could be to annotate `GlobalAlloc` impls in which unwinding is Undefined Behavior.
+
+<!-- lint-group-start: panicking -->
+Lints: `arithmetic_side_effects`, `expect_used`, `unwrap_used`, `panic`, `unreachable`, `todo`, `unimplemented`, `string_slice`, `indexing_slicing`
+
+<details>
+<summary>#![warn] attribute</summary>
+
+```
+#![warn(
+    clippy::arithmetic_side_effects,
+    clippy::expect_used,
+    clippy::unwrap_used,
+    clippy::panic,
+    clippy::unreachable,
+    clippy::todo,
+    clippy::unimplemented,
+    clippy::string_slice,
+    clippy::indexing_slicing
+)]
+```
+</details>
+
+<details>
+<summary>Lint table</summary>
+
+```
+[lints.clippy]
+arithmetic_side_effects = "warn"
+expect_used = "warn"
+unwrap_used = "warn"
+panic = "warn"
+unreachable = "warn"
+todo = "warn"
+unimplemented = "warn"
+string_slice = "warn"
+indexing_slicing = "warn"
+```
+</details>
+<!-- lint-group-end: panicking -->
+
+## Debugging
+
+These are lints that can be useful to disable in CI, as they might indicate that code needs more work
+or has remaining debugging artifacts.
+
+<!-- lint-group-start: debugging -->
+Lints: `dbg_macro`, `todo`, `unimplemented`
```

Comment:
[Per `unimplemented!`'s docs](https://doc.rust-lang.org/std/macro.unimplemented.html):
> The difference between `unimplemented!` and `todo!` is that while `todo!` conveys an intent of implementing the functionality later and the message is “not yet implemented”, `unimplemented!` makes no such claims. Its message is “not implemented”.

By omission, this implies that `unimplemented!` can be used to indicate intentionally/permanently unimplemented behaviour (eg. for target specific code), which implies that its use is not limited to debugging.

Reply:
Right, good point. I'll remove that one later